### PR TITLE
CCAIA-877 Preserve MDC context in Session repository

### PR DESCRIPTION
**Pull request by reddy-yattapu**

### CODE REVIEW

**Comment by darrell-plant** [comment]
LGTM 👍

File: app/repositories/SessionRepository.scala [hunk 1]
```diff
@@ -57,43 +59,49 @@ class SessionRepository @Inject()(formId: String, mongo: ReactiveMongoComponent)
...
+        sessionId <- getSessionId
+        maybeOption <- findById(sessionId)
+      } yield {
+        maybeOption
+          .map(_.data \ formId)
+          .flatMap(x =>
+            x match {
+              case JsDefined(value) => Some(value.as[A])
+              case JsUndefined() => None
+            })
```
Comment (PeterPerhac) [inline app/repositories/SessionRepository.scala]: I know you didn't write this originally, but it struck me as particularly ugly piece of code

```suggestion
          maybeOption.flatMap(sd => (sd.data \ formId).asOpt[A])
```

**Comment by PeterPerhac** [comment]
👍 LGTM

**Comment by jon-pike** [comment]
👍

**Pull request merged**